npm/private/npm_translate_lock.bzl

```diff
@@ -987,10 +984,27 @@ load("@aspect_rules_js//npm/private:npm_package_store.bzl", _npm_package_store =
         visibility = ["//visibility:public"],
     )""")
 
-    rctx.file(_DEFS_BZL_FILENAME, "\n".join(defs_bzl_header + [""] + defs_bzl_body + [""]))
-    rctx.file(_REPOSITORIES_BZL_FILENAME, "\n".join(repositories_bzl))
+    rctx_files[_DEFS_BZL_FILENAME] = ["\n".join(defs_bzl_header + [""] + defs_bzl_body + [""])]
+    rctx_files[_REPOSITORIES_BZL_FILENAME] = ["\n".join(repositories_bzl)]
+
+    for filename, contents in rctx.attr.additional_file_contents.items():
+        if not filename in rctx_files.keys():
+            rctx_files[filename] = contents
+        elif filename[len(filename) - 4:] == ".bzl":
```

Comment:
There is an [endswith](https://bazel.build/rules/lib/string#endswith) method?

Reply:
🤦